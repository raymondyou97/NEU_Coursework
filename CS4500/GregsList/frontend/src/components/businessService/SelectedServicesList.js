import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';

const SelectedServicesList = ({
  selectedServices,
  setSelectedServices,
  selectedSelectedService,
  setSelectedSelectedService,
}) => {
  if (selectedServices) {
    return (
      <ListGroup>
        {selectedServices.map(({ id, title }) => {
          const isSelected = selectedSelectedService === id;
          return (
            <ListGroup.Item
              key={id}
              active={isSelected}
              onClick={() =>
                setSelectedSelectedService(
                  id === selectedSelectedService ? undefined : id
                )
              }
            >
              {title}
              <button
                onClick={() => {
                  setSelectedServices(
                    selectedServices.filter(x => x.id !== id)
                  );
                  setSelectedSelectedService(
                    isSelected ? undefined : selectedSelectedService
                  );
                }}
              >
                {'X'}
              </button>
            </ListGroup.Item>
          );
        })}
      </ListGroup>
    );
  }

  return null;
};

SelectedServicesList.propTypes = {
  selectedServices: PropTypes.arrayOf(PropTypes.object.isRequired),
  selectedSelectedService: PropTypes.number,
  setSelectedSelectedService: PropTypes.func.isRequired,
};

export default SelectedServicesList;
