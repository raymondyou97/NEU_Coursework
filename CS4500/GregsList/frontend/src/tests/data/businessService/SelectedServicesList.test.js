import React from 'react';
import renderer from 'react-test-renderer';

import SelectedServicesList from '../../../components/businessService/SelectedServicesList';
import data from './selected-services.mock.json';

test('SelectedServicesList Snapshot renders', () => {
  const component = renderer.create(
    <SelectedServicesList
      selectedServices={data}
      selectedSelectedServices={() => {}}
      selectedSelectedService={data[0]['id']}
      setSelectedSelectedService={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
