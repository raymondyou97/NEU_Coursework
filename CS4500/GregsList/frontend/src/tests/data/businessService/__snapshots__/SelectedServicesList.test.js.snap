// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedServicesList Snapshot renders 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item active"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    Appliance Installation
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    Cabinetry
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    Door Installation
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    Concrete Removal
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    Dishwasher Installation
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
</div>
`;
